pyxrf/model/draw_image_rgb.py

```diff
@@ -290,17 +309,69 @@ def _compute_equal_axes_ranges(x_min, x_max, y_min, y_max):
-        if (xd <= math.floor(yd / 100)) and (yd >= 200):
-            xd_min, xd_max = -math.floor(yd / 200), math.ceil(yd / 200)
+        def _adjust_data_range__min_ratio(c_min, c_max, c_axis_range, *, min_ratio=0.01):
+            """
+            Adjust the range for plotted data along one axis (x or y). The adjusted range is
+            applied to the 'extend' attribute of imshow(). The adjusted range is always greater
+            than 'axis_range * min_ratio'. Such transformation has no physical meaning
+            and performed for aesthetic reasons: stretching the image presentation of
+            a scan with only a few lines (1-3) greatly improves visibility of data.
 
-        xd_axis_min, xd_axis_max, yd_axis_min, yd_axis_max = \
-            _compute_equal_axes_ranges(xd_min, xd_max, yd_min, yd_max)
+            Parameters
+            ----------
+
+            c_min, c_max : float
+                boundaries of the data range (along x or y axis)
+            c_axis_range : float
+                range presented along the same axis
+
+            Returns
+            -------
+
+            cmin, c_max : float
+                adjusted boundaries of the data range
+            """
+            c_range = c_max - c_min
+            if c_range < c_axis_range * min_ratio:
+                c_center = (c_max + c_min) / 2
+                c_new_range = c_axis_range * min_ratio
+                c_min = c_center - c_new_range / 2
+                c_max = c_center + c_new_range / 2
+            return c_min, c_max
+
+        if self.pixel_or_pos:
+
+            # xd_min, xd_max, yd_min, yd_max = min(self.x_pos), max(self.x_pos),
+            #     min(self.y_pos), max(self.y_pos)
+            x_pos_2D = self.data_dict['positions']['x_pos']
+            y_pos_2D = self.data_dict['positions']['y_pos']
+            xd_min, xd_max, yd_min, yd_max = x_pos_2D.min(), x_pos_2D.max(), y_pos_2D.min(), y_pos_2D.max()
+            xd_axis_min, xd_axis_max, yd_axis_min, yd_axis_max = \
+                _compute_equal_axes_ranges(xd_min, xd_max, yd_min, yd_max)
+
+            xd_min, xd_max = _adjust_data_range__min_ratio(xd_min, xd_max, xd_axis_max - xd_axis_min)
+            yd_min, yd_max = _adjust_data_range__min_ratio(yd_min, yd_max, yd_axis_max - yd_axis_min)
+
+            # Adjust the direction of each axis depending on the direction in which encoder values changed
+            #   during the experiment. Data is plotted starting from the upper-right corner of the plot
+            if x_pos_2D[0, 0] > x_pos_2D[0, -1]:
+                xd_min, xd_max, xd_axis_min, xd_axis_max = xd_max, xd_min, xd_axis_max, xd_axis_min
+            if y_pos_2D[0, 0] > y_pos_2D[-1, 0]:
+                yd_min, yd_max, yd_axis_min, yd_axis_max = yd_max, yd_min, yd_axis_max, yd_axis_min
+
+        else:
+
+            # Set equal ranges for the axes data
+            yd, xd = selected_data.shape[1], selected_data.shape[2]
```

Comment:
Is it a 3d array? What's the 0th dimension?

Reply:
Yes, it is 3d array. The 0th dimension is the number of the image (RGB color):

Lines 376-381 of the same file:

```
        name_r = self.rgb_name_list[self.index_red]
        data_r = selected_data[self.index_red, :, :]
        name_g = self.rgb_name_list[self.index_green]
        data_g = selected_data[self.index_green, :, :]
        name_b = self.rgb_name_list[self.index_blue]
        data_b = selected_data[self.index_blue, :, :]
```

Reply:
OK, thanks for the explanation.